process/block/baseProcess.go

```diff
@@ -255,7 +255,9 @@ func (bp *baseProcessor) requestHeadersIfMissing(
 		missingNonces = append(missingNonces, nonces...)
 	}
 
+	headersPool := bp.dataPool.Headers()
```

Comment:
some tests for the added code would be nice.

Reply:
would be added until the system tests are done

Reply:
great